sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -100,6 +102,75 @@ def test_unsupervised_inputs():
         assert_array_almost_equal(ind1, ind2)
 
 
+def test_precomputed():
+    """Tests unsupervised NearestNeighbors with a distance matrix."""
+    # Note: smaller samples may result in spurious test success
```

Comment:
We need to use a random number generator created in this test, to avoid having side effects across tests.